### PR TITLE
Add tacotron model

### PR DESCRIPTION
**Pull request by jimchen90**
This is to add the first part of tacotron2 model.

### CODE REVIEW

**Comment by vincentqb** [comment]
It's the first part of tacotron 2, right? and it builds on top of #835 and #844, right?

**Comment by jimchen90** [comment]
> It's the first part of tacotron 2, right? and it builds on top of #835 and #844, right?

Yes, this is the first part of tacotron 2. And it is a separate one.

File: torchaudio/models/_tacotron2.py [hunk 1]
```diff
@@ -0,0 +1,1048 @@
...
+
+        x = nn.utils.rnn.pack_padded_sequence(x, input_lengths, batch_first=True)
+
+        self.lstm.flatten_parameters()
+        outputs, _ = self.lstm(x)
+        outputs, _ = nn.utils.rnn.pad_packed_sequence(outputs, batch_first=True)
+
+        return outputs
+
+    def infer(self, x: Tensor, input_lengths: Tensor) -> Tensor:
```
Comment (vincentqb) [inline torchaudio/models/_tacotron2.py:457]: What's the difference with forward here?

File: torchaudio/models/_tacotron2.py [hunk 2]
```diff
@@ -0,0 +1,1048 @@
...
+            gate_outputs += [gate_output.squeeze()]
+            alignments += [attention_weights]
+
+        mel_outputs, gate_outputs, alignments = self._parse_decoder_outputs(
+            torch.stack(mel_outputs), torch.stack(gate_outputs), torch.stack(alignments)
+        )
+
+        return mel_outputs, gate_outputs, alignments
+
+    def infer(self, memory: Tensor, memory_lengths: Tensor) -> Tuple[Tensor]:
```
Comment (vincentqb) [inline torchaudio/models/_tacotron2.py:786]: same

File: torchaudio/models/_tacotron2.py [hunk 3]
```diff
@@ -0,0 +1,1048 @@
...
+        )
+
+        mel_outputs_postnet = self.postnet(mel_outputs)
+        mel_outputs_postnet = mel_outputs + mel_outputs_postnet
+
+        return self.parse_output(
+            [mel_outputs, mel_outputs_postnet, gate_outputs, alignments], output_lengths
+        )
+
+    def infer(self, inputs: Tensor, input_lengths: Tensor) -> Tuple[Tensor]:
```
Comment (vincentqb) [inline torchaudio/models/_tacotron2.py:1033]: same

File: torchaudio/models/_tacotron2.py (hunk 1 above)
Comment (jimchen90) [reply torchaudio/models/_tacotron2.py:457]: The difference here is the dropout layer. I agree, since the "infer" won't be used in the training, I will remove this part.

**Comment by yangarbiter** [comment]
Continuing this feature in #1621 .
